indico/modules/events/management/controllers.py

```diff
@@ -328,6 +400,89 @@ def _process(self):
         return jsonify_data(flash=False)
 
 
+class RHClonePreview(RHManageEventBase):
+    def _process(self):
+        form = CloneRepeatabilityForm()
+
+        if form.repeatability.data == 'interval':
+            clone_calculator = IntervalCloneCalculator(self.event_new)
+        elif form.repeatability.data == 'pattern':
+            clone_calculator = PatternCloneCalculator(self.event_new)
+        else:
+            raise BadRequest
+
+        try:
+            dates = clone_calculator.calculate(request.form)
+            if len(dates) > 100:
+                raise ValueError(_("You can clone maximum of 100 times at once"))
+        except ValueError as e:
+            return jsonify(error={'message': e.message})
+        return jsonify_data(count=len(dates), dates=dates)
```

Comment:
`flash=False` or, if this is not processed by ajaxdialog/ajaxform logic, simply `jsonify()`